lang/terraform/terraform.py

```diff
@@ -0,0 +1,156 @@
+from talon import Context, Module, actions, settings
+
+ctx = Context()
+mod = Module()
+ctx.matches = r"""
+mode: user.terraform
+mode: user.auto_lang
+and code.language: terraform
+"""
+
+types = {
+    "string": "string",
+    "number": "number",
+    "bool": "bool",
+    "list": "list",
+    "map": "map",
+    "null": "null"
+}
+
+ctx.lists["user.code_type"] = types
+
+common_properties = {
+    "name": "name",
+    "type": "type",
+    "description": "description",
+    "default": "default",
+    "for each": "for_each",
+    "count": "count",
+    "prevent destroy": "prevent_destroy",
+    "nullable": "nullable",
+    "sensitive": "sensitive",
+    "depends on": "depends_on",
+    "provider": "provider",
+    "source": "source",
+}
+
+mod.list("terraform_common_property", desc="Terraform Modifier")
+ctx.lists["self.terraform_common_property"] = common_properties
+
+module_blocks = {
+    "variable": "variable",
+    "output": "output",
+    "provider": "provider",
+    "module": "module",
+}
+
+mod.list("terraform_module_block", desc="Simple Terraform Block")
+ctx.lists["self.terraform_module_block"] = module_blocks
+
+
+@mod.action_class
+class Actions:
+
+    def code_terraform_module_block(text: str):
+        """Inserts a new module-related block of a given type (e.g. variable, output, provider...)"""
+
+    def code_terraform_resource(text: str):
+        """Inserts a new resource block with given name"""
+
+    def code_terraform_data_source(text: str):
+        """Inserts a new data block with given name"""
+
+
+@ctx.action_class("user")
+class UserActions:
+
+    def code_terraform_module_block(text: str):
+        actions.auto_insert(text + " \"\"")
+        actions.key("left")
+
+    def code_terraform_resource(text: str):
+        result = "resource \"{}\" \"\"".format(
```

Comment:
nit: use single quotes to avoid escaping and make this more readable
```suggestion
        result = 'resource "{}" ""'.format(
```